lib/confluence/aggressive_removal.es6.js

```diff
@@ -0,0 +1,144 @@
+        },
+      ],
+      code: function(browsers, date) {
+        for (let i = 0; i < browsers.length; i++) {
+          let browser = browsers[i];
+          let year = date.getFullYear();
+          let month = date.getMonth();
+          let day = date.getDate();
+          let oneYearAgo = new Date(year - 1, month, day);
+          let browserKeys = browsers.map((browser) => browser.browserKey);
+          let selectBrowsers = [
+            this.browserDAO.where(this.mlang.AND(
+              this.mlang.IN(this.Browser.BROWSER_KEY, browserKeys),
+              this.mlang.NEQ(this.Browser.BROWSER_NAME, browser.browserName)))
+            .select(),
+            this.browserDAO.where(this.mlang.AND(
+              this.mlang.EQ(this.Browser.BROWSER_NAME, browser.browserName),
+              this.mlang.EQ(this.Browser.OS_NAME, browser.osName),
+              this.mlang.LT(this.Browser.RELEASE_DATE, oneYearAgo)))
+            .select()];
+          Promise.all(selectBrowsers).then((result) => {
+            let currBrowsers = result[0].a;
+            let prevReleaseBrowsers = result[1].a;
+            // Return if there is insufficient of historical data.
+            if (prevReleaseBrowsers.length < 2) return;
+            let browser = prevReleaseBrowsers.pop();
+            let removedIfaces = {};
+            let promises = [];
+            // Find all APIs available this browser between date - 2years
+            // to date - 3years period.
+            for (let i = 0; i < prevReleaseBrowsers.length; i++) {
+              promises.push(prevReleaseBrowsers[i].interfaces.select({
+                put: function(iface) {
+                  if (!removedIfaces.hasOwnProperty(iface.interfaceKey)) {
+                    removedIfaces[iface.interfaceKey] = 0;
+                  }
+                },
+              }));
+            }
+            // Find removed APIs. (APIs in browsers between date - 2years
+            // to date - 3years set minus APIs in version released just
+            // before date -2years).
+            Promise.all(promises).then(() => {
+              browser.interfaces.select({
+                put: function(iface) {
+                  if (removedIfaces.hasOwnProperty(iface.interfaceKey)) {
+                    delete removedIfaces[iface.interfaceKey];
+                  }
+                },
+              }).then(() => {
+                // Find the interfaces that still exists in today's
+                // other browsers.
+                promises = [];
+                for (let i = 0; i < currBrowsers.length; i++) {
+                  promises.push(currBrowsers[i].interfaces.select({
+                    put: function(iface) {
+                      if (removedIfaces.hasOwnProperty(iface.interfaceKey)) {
+                        removedIfaces[iface.interfaceKey]++;
+                      }
+                    },
+                  }));
+                }
+                Promise.all(promises).then(() => {
```

Comment:
No, but this is `compute()`, not `init()`. We may later restructure the code to perform `compute()` from another call site that wants to wait for the computation to complete.